src/java/org/apache/cassandra/db/compaction/UnifiedCompactionStrategy.java

```diff
@@ -459,32 +532,43 @@ private static double getRangesTotalSize(List<Splitter.WeightedRange> ranges)
                 // Note: We allow empty aggregates into the list of pending compactions. The pending compactions list
                 // is for progress tracking only, and it is helpful to see empty levels there.
                 pending.add(aggregate);
-
-                // Make sure the level count includes all levels for which we have sstables (to be ready to compact
-                // as soon as the threshold is crossed)...
-                levelCount = Math.max(levelCount, aggregate.bucketIndex() + 1);
-                if (aggregate.selected != null)
-                {
-                    // ... and also the levels that a layout-preserving selection would create.
-                    levelCount = Math.max(levelCount, levelOf(aggregate.selected) + 1);
-                }
-
-                // The space overhead limit also applies when a single compaction is above that limit. This should
-                // prevent running out of space at the expense of several highest-level tables extra, i.e. slightly
-                // higher read amplification, which I think is a sensible tradeoff; however, operators must be warned
-                // if this happens.
-                warnIfSizeAbove(aggregate, spaceOverheadLimit);
             }
         }
 
         // Update the tracked background tasks.
         backgroundCompactions.setPending(this, pending);
+        return pending;
+    }
+
+    /**
+     * Selects compactions to run next from the passed aggregates.
+     *
+     * The intention here is to use this method directly from outside processes, to run compactions from a set
+     * of pre-existing aggregates, that have been generated out of process.
+     *
+     * @param aggregates
+     * @return a subset of compaction aggregates to run next
+     */
+    public Collection<CompactionAggregate> getNextCompactionAggregates(Collection<CompactionAggregate.UnifiedAggregate> aggregates, int maxConcurrentCompactions)
```

Comment:
This method is very similar to its overload and furthermore, this method doesn't look correct. I believe the two method overloads need to be refactored to have the same code be only in one commonly used method.

Reply:
I have made some refactoring that I hope will be an improvement, however there are differences that I think prevent a single common method.